editprofile.js

```diff
@@ -0,0 +1,240 @@
+      users[currentUserIndex].wineColors.push(checkbox.id)
+      console.log(checkbox.id);
+    } else {
+      if (checkbox.checked) {
+        console.log('wont work');
+      }
+    }
+  }
+
+  for (i = 0; i< wineList.length; i++){
+    checkbox = form[wineList[i]];
+    console.log(checkbox.id);
+    if ((checkbox.id != users[currentUserIndex].favWines[i]) && checkbox.checked) {
+      checkbox = form[wineList[i]];
+      users[currentUserIndex].favWines.push(checkbox.id)
+      console.log(checkbox.id);
+    } else {
+      if (checkbox.checked) {
+        console.log('wont work');
+      }
+    }
+  }
+
+  for (i = 0; i< wineries.length; i++){
+    checkbox = form[wineries[i]];
+    if ((checkbox.id != users[currentUserIndex].favWineries[i]) && checkbox.checked) {
+      checkbox = form[wineries[i]];
+      users[currentUserIndex].favWineries.push(checkbox.id)
+      console.log(checkbox.id);
+    } else {
+      if (checkbox.checked) {
+        console.log('wont work');
+      }
+
+  }
+}
+
+  try {
+    localStorage.users = JSON.stringify(users);
+  } catch (error){
+    console.log('failed to save updated list');
+
+  }
+}
+//end update favs lists handler
+function handleSubmitReview (event) {
+
+  event.preventDefault();
+  var form = event.target;
+  var reviewUpdate = form.userWineReview.value;
+  console.log(reviewUpdate);
+  var section = document.getElementById('reviewBox');
+  var pTag = document.createElement('p');
+  section.appendChild(pTag);
+  pTag.textContent = (reviewUpdate);
+  users[currentUserIndex].reviews.push(reviewUpdate);
+
+try {
+  localStorage.users = JSON.stringify(users);
+  console.log('users', users);
+}
+catch (error) {
+  console.log('something went wrong', error);
+}
+form.reset();
+genorateReview();
+}
+function genorateReview(){
+  for (var i = 0; i <= users[currentUserIndex].reviews.length; i++) {
+    var unordered = document.getElementById('reviewBox');
+  var list = document.createElement('ul');
+  var listItem = document.createElement('li');
+  listItem.textContent =
+  (users[currentUserIndex].reviews[i]);
+  console.log(users[currentUserIndex].reviews[i])
+  unordered.appendChild(list);
+  list.appendChild(listItem);
+}
+}
+//
+function handleUserBio (event){
+  event.preventDefault();
+  usersBio.textContent = '';
+  var form = event.target;
+  var userBioUpdate = form.userBio.value;
+  console.log(userBioUpdate);
+  users[currentUserIndex].bio = userBioUpdate;
+  form.reset();
+  populateBio();
+  try {
+    localStorage.users = JSON.stringify(users);
+    console.log('users', users);
+  }
+  catch (error) {
+    console.log('something went wrong', error);
+  }
+
+}
+function populateBio(){
+  var bioUpdate = document.getElementById('usersBio');
+  var text = document.createTextNode(users[currentUserIndex].bio);
+  var list = document.createElement('li');
+  list.setAttribute('id', 'bio-list');
+  bioUpdate.appendChild(list);
+  list.appendChild(text);
+}
+populateBio();
```

Comment:
it would be easier to read thif file if all code that run was
at the bottom :)

try organizing your code in the following order
globals
func defs
event handlers
run code